src/timing.h

```diff
@@ -110,11 +111,7 @@ void jl_timing_printf(jl_timing_block_t *cur_block, const char *format, ...);
 }
 #endif
 
-#ifdef __cplusplus
-#define JL_TIMING_CURRENT_BLOCK (&__timing_block.block)
-#else
-#define JL_TIMING_CURRENT_BLOCK (&__timing_block)
-#endif
+#define JL_TIMING_CURRENT_BLOCK (jl_current_task->ptls->timing_stack)
```

Comment:
It'd be nice to return the timing block from the `JL_TIMING` macro, so that we can avoid repeated lookups in thread-local storage (which is relatively slow on platforms such as macOS)

Otherwise this change is good, because it means that we can easily add metadata from inner functions.

Reply:
I needed to change this because otherwise we'd end up with duplicate `__timing_block` variables when conditional macros are used. I guess I could maybe do some concatenation of the event name, and then change the macro to take owner/event to identify the correct variable.

Reply:
I was thinking that you'd get the timing_block from the `JL_TIMING` macro and leave `JL_TIMING_CURRENT_BLOCK` as you have it here, but use it only when you don't have the block on-hand and need to look it up again

Reply:
I don't think there's a standard C-friendly way to return a value from a multi-statement macro, I think we will just end up having something like __timing_block_OWNER_EVENT inside JL_TIMING, and then access it via JL_TIMING_BLOCK(OWNER, EVENT).

Reply:
Yeah, I prefer your version over a macro that introduces locals - looks good to me